r/tests/testthat/test-dplyr-funcs-datetime.R

```diff
@@ -711,3 +711,34 @@ test_that("am/pm mirror lubridate", {
   )
 
 })
+
+test_that("extract tz", {
+  df <- tibble(
+    posixct_date = as.POSIXct(c("2022-02-07", "2022-02-10"), tz = "Pacific/Marquesas"),
+  )
+
+  compare_dplyr_binding(
+    .input %>%
+      mutate(timezone_posixct_date = tz(posixct_date)) %>%
+      collect(),
+    df
+  )
+
+  expect_error(
+    call_binding("tz", Expression$scalar("2020-10-01")),
+    "timezone extraction for objects of class `string` not supported in Arrow"
+  )
```

Comment:
The bit that I think you were missing is `Expressoin$scalar(...)`. I found this by looking around to find similar calls in the tests.  Though there aren't any that are exactly the same, I found 

https://github.com/apache/arrow/blob/3a8e409385c8455e6c80b867c5730965a501d113/r/tests/testthat/test-dplyr-funcs-datetime.R#L113-L120

which lead me to look at `Expression`:

https://github.com/apache/arrow/blob/3a8e409385c8455e6c80b867c5730965a501d113/r/R/expression.R#L188-L192